sdk/identity/azure-identity/src/managed_identity_source.cpp

```diff
@@ -250,20 +250,28 @@ std::unique_ptr<ManagedIdentitySource> AppServiceV2019ManagedIdentitySource::Cre
       credName, clientId, resourceId, options, "IDENTITY_ENDPOINT", "IDENTITY_HEADER", "2019");
 }
 
-// Cloud Shell doesn't support user-assigned managed identities
 std::unique_ptr<ManagedIdentitySource> CloudShellManagedIdentitySource::Create(
     std::string const& credName,
     std::string const& clientId,
-    std::string const&,
+    std::string const& resourceId,
     Azure::Core::Credentials::TokenCredentialOptions const& options)
 {
+  using Azure::Core::Credentials::AuthenticationException;
+
   constexpr auto EndpointVarName = "MSI_ENDPOINT";
   auto msiEndpoint = Environment::GetVariable(EndpointVarName);
 
   std::string const CredSource = "Cloud Shell";
 
   if (!msiEndpoint.empty())
   {
+    if (!clientId.empty() || !resourceId.empty())
+    {
+      throw AuthenticationException(
+          "User-assigned managed identities are not supported in Cloud Shell environments. Omit "
+          "the client or resource ID when constructing the ManagedIdentityCredential.");
```

Comment:
```suggestion
          "the clientId or resourceId when constructing the ManagedIdentityCredential.");
```

I don't feel strongly if it should be `clientId` and `resourceId` or if it should be `client id` and `resource id`, but since the two parameters follow the same naming pattern, their names in the exception should follow a similar naming pattern.

Reply:
Id is a strange word, I never know how to write it. It is not an abbreviation (I.D.), it is not an "id" from psychoanalysis (as in "id software", pronounced "eed"). You want to write it with both capital letters, i.e. ID, yet it is not an acronym. However, if you write it this way, everyone would pronounce it correctly, and it won't be mistaken from psychoanalysis' id. I guess, it is the right way to write it then - both capital letters, no periods. Everything is fine then, besides the fact that no other word was formed this way, but it does not mean that "ID" has no right to exist as a word. I think, someone who would invent an acronym for this, it would close this hole, and the word will be just like all others.

On a side note, it is also fascinating, how capitalized and non-capitalized letters do matter in post-Latin languages, but in the original Latin, there was no notion of uppercase and lowercase letters, there were just 20+ letters, and European languages seem to be the only ones making distinction between letter cases - "A" and "a" are kind of the same letter, but not completely.

Reply:
My suggestion, match the identity variable name `client_id` and `resource_id`